packages/flutter_tools/lib/src/build_system/targets/deferred_components.dart

```diff
@@ -0,0 +1,107 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:meta/meta.dart';
+
+import '../../android/deferred_components_gen_snapshot_validator.dart';
+import '../../base/deferred_component.dart';
+import '../../build_info.dart';
+import '../../project.dart';
+import '../build_system.dart';
+import '../depfile.dart';
+import 'android.dart';
+
+/// Creates a [DeferredComponentsSetupValidator and displays the validator
+/// output to the developer if changes are recommended.
+class DeferredComponentsGenSnapshotValidatorTarget extends Target {
+  /// Create an [AndroidAotDeferredComponentsBundle] implementation for a given [targetPlatform] and [buildMode].
+  DeferredComponentsGenSnapshotValidatorTarget({
+    this.dependency,
+    this.title,
+    this.exitOnFail = true,
+    String name = 'deferred_components_setup_validator',
+  }) : _name = name;
+
+  /// The [AndroidAotDeferredComponentsBundle] derived target instances this rule depends on.
+  final CompositeTarget dependency;
+
+  /// The title of the [DeferredComponentsSetupValidator] that is
+  /// displayed to the developer when logging results.
+  final String title;
+
+  /// Whether to exit the tool if a recommended change is found by the
+  /// [DeferredComponentsSetupValidator].
+  final bool exitOnFail;
+
+  /// The name of the produced Android ABI.
+  List<String> get _androidAbiNames {
+    final List<String> abis = <String>[];
+    if (dependency == null) {
+      return abis;
+    }
+    for (final Target target in dependency.dependencies) {
+      if (target.dependencies.isNotEmpty) {
+        abis.add(getNameForAndroidArch(
+          getAndroidArchForName(getNameForTargetPlatform((target.dependencies.first as AndroidAotBundle).dependency.targetPlatform))));
```

Comment:
If this target requires a specific type I think it would be better to accept them directly instead of having runtime checks here

Reply:
I'll have the abis passed in as a parameter directly so we can remove the dependency on specific types of targets.